build_scripts/run_tests.py

```diff
@@ -52,62 +56,81 @@ def print_report(self):
     num_failures = len(self._failures)
     if num_failures == 0:
       return
-    print("\n%d module(s) failed: \n" % num_failures)
+    print("\n%d test module(s) failed: \n" % num_failures)
     for mod_name, log_file in self._failures:
       print("** %s - %s" % (mod_name, log_file))
 
 
-def run_ninja(targets, fail_collector):
+def run_ninja(targets, fail_collector, fail_fast=False):
   """Run ninja over the list of specified targets.
 
   Arguments:
     targets: The list of test targets to run.
     fail_collector: A FailCollector object to collect failures.
+    fail_fast: If True, abort at the first target failure.
 
   Returns:
     True if no target fails. False, otherwise.
   """
   # The -k option to ninja, set to a very high value, makes it run until it
-  # detects all failures.
-  cmd = ["ninja", "-k", "100000"] + targets
+  # detects all failures. So, we set it to a high value unless |fail_fast| is
+  # True.
+  cmd = ["ninja", "-k", "1" if fail_fast else "100000"] + targets
   process = subprocess.Popen(cmd, cwd=build_utils.OUT_DIR,
                              stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-  have_failures = False
-  while True:
-    # process.stdout.readline() always returns a 'bytes' object.
-    l = process.stdout.readline().decode("utf-8")
-    if not l:
-      break
-    mod_name, log_file = test_module.get_module_and_log_file_from_result_msg(l)
-    if mod_name:
-      print(l)
-    if log_file:
-      have_failures = True
-      if fail_collector:
-        fail_collector.add_failure(mod_name, log_file)
+  failed_targets = []
+  with open(build_utils.NINJA_LOG, "w") as ninja_log:
+    while True:
+      l = process.stdout.readline()
```

Comment:
nit: lowercase L looks like number 1; maybe call this `line` or something?

Reply:
Done.